resources/js/components/dashboard/components/IndividualSponsor.tsx

```diff
@@ -0,0 +1,351 @@
+
+interface IIndividualSponsorProps {
+    SponsorId: string,
+}
+
+interface IIndividualSponsorState {
+    SponsorId: number | undefined,
+    loading: boolean,
+    Sponsor: ISponsor | undefined,
+    user: IUserDetails | undefined,
+    loadingDefinitions: boolean,
+    resources: undefined,
+    totalNo: number,
+    portalInfo: undefined,
+}
+
+const sponsorFields: { id: string, title: string, placeholder: string }[] = [
+    { id: "1", title: "Industry", placeholder: "" },
+    { id: "2", title: "Homepage", placeholder: "Mention anything you want -- it doesn’t have to be technology-related!" },
+    { id: "3", title: "Description", placeholder: "" },
+    { id: "4", title: "Opportunities", placeholder: "For example GitHub, LinkedIn or your website. Put each link on a new line. " },
+]
+
+class IndividualSponsor extends Component<IIndividualSponsorProps & RouteComponentProps, IIndividualSponsorState> {
+    
+    state = {
+        SponsorId: undefined, // possibly redundant copy of ID
+        loading: true, // loading main spons object
+        loadingDefinitions: true, // loading additional resource cards
+        Sponsor: undefined, // key sponsor object
+        user: undefined,
+        resources: undefined, //where all resource cards are stored
+        totalNo: 0, //for pagination purposes
+        portalInfo: {
+            data:{
+                ["description"]:"",
+                ["url"]:"",
+                ["discord invite link"]:"",
+            },
+            files:[],
+        }, //for company main info purposes
+    }
+
+    constructor(props: IIndividualSponsorProps & RouteComponentProps){
+        super(props);
+    }
+
+    componentDidMount() {
+        const SponsorId = +this.props.SponsorId;
+        if (Number.isNaN(SponsorId)) {
+            this.setState({ loading: false });
+        } else{
+            this.retrieveSponsor(SponsorId);
+        }
+    }
+
+    componentDidUpdate(){
+        if (this.state.Sponsor && this.state.loadingDefinitions === true){
+            this.loadInformation();
+        }
+    }
+
+    render() {
+        const { SponsorId, loading } = this.state;
+        return (<>
+            { loading 
+                ? this.loadingMarkup
+                : SponsorId 
+                    ? this.renderSponsor()
+                    : this.invalidSponsor()
+            } 
+        </>);
+    }
+
+    private loadingMarkup = <>
+        <Page title={"Loading..."}>
+            <Card sectioned>
+                <SkeletonBodyText />
+            </Card>
+        </Page>
+    </>;
+
+    private nextSponsor = () => {
+        this.setState({ loading: true });
+        const total = this.state.totalNo;
+        const currentUrl = this.props.history.location.pathname;
+        const base = currentUrl.substring(0, currentUrl.lastIndexOf('/'));
+        const currentSpons:string = currentUrl.substring(currentUrl.lastIndexOf('/')+1);
```

Comment:
`currentSpons` is just weird, I would prefer the full `currentSponsor` (you are saving only 2 characters 😅 )

Reply:
Changed